tests/unit/test_stimulus_manager.py

```diff
@@ -0,0 +1,881 @@
+        "Mode": "Current",
+        "Duration": 2,
+        "Delay": 0,
+        "RiseTime": 0.1,
+        "DecayTime": 0.5,
+        "MeanPercent": 20,
+        "SDPercent": -0.5,
+    }
+    with pytest.raises(Exception, match="RelativeShotNoise stdev percent must be positive"):
+        ringtest_stimulus_manager.interpret(target_onecell, stim_info)
+
+    stim_info = {
+        "Pattern": "RelativeShotNoise",
+        "Mode": "Current",
+        "Duration": 2,
+        "Delay": 0,
+        "RiseTime": 0.1,
+        "DecayTime": 0.5,
+        "MeanPercent": 20,
+        "SDPercent": 0.5,
+        "RelativeSkew": 1.5,
+    }
+    with pytest.raises(Exception, match=r"RelativeShotNoise relative skewness must be in \[0,1\]"):
+        ringtest_stimulus_manager.interpret(target_onecell, stim_info)
+
+
+def test_error_absolute_shot_noise(ringtest_stimulus_manager):
+    """Test various checks on AbsoluteShotNoise parameters"""
+
+    stim_info = {
+        "Pattern": "AbsoluteShotNoise",
+        "Mode": "Current",
+        "Duration": 2,
+        "Delay": 0,
+        "Dt": 0.5,
+        "RiseTime": 0.1,
+        "DecayTime": 0.5,
+        "Mean": 10,
+        "Sigma": -1,
+    }
+    with pytest.raises(Exception, match="AbsoluteShotNoise stdev must be positive"):
+        ringtest_stimulus_manager.interpret(target_onecell, stim_info)
+
+    stim_info = {
+        "Pattern": "AbsoluteShotNoise",
+        "Mode": "Current",
+        "Duration": 2,
+        "Delay": 0,
+        "Dt": 0.5,
+        "RiseTime": 0.1,
+        "DecayTime": 0.5,
+        "Mean": 10,
+        "Sigma": 1,
+        "RelativeSkew": 1.5,
+    }
+    with pytest.raises(Exception, match=r"AbsoluteShotNoise relative skewness must be in \[0,1\]"):
+        ringtest_stimulus_manager.interpret(target_onecell, stim_info)
+
+
+def test_warning_ornsteinuhlenbeck(caplog, ringtest_stimulus_manager):
+    """Warning on negativte mean + abs(mean)<2*sigma"""
+
+    stim_info = {
+        "Pattern": "OrnsteinUhlenbeck",
+        "Mode": "Current",
+        "Duration": 2,
+        "Delay": 0,
+        "Dt": 0.5,
+        "Tau": 2.7,
+        "Reversal": 0.1,
+        "Mean": -10,
+        "Sigma": 1,
+    }
+    with caplog.at_level(logging.WARNING):
+        ringtest_stimulus_manager.interpret(target_onecell, stim_info)
+        assert "OrnsteinUhlenbeck signal is mostly zero" in caplog.text
+
+
+def test_warning_relative_ornsteinuhlenbeck(caplog, ringtest_stimulus_manager):
+    """Warning on negativte mean + abs(mean)<2*sigma"""
+
+    stim_info = {
+        "Pattern": "RelativeOrnsteinUhlenbeck",
+        "Mode": "Current",
+        "Duration": 2,
+        "Delay": 0,
+        "Dt": 0.5,
+        "Tau": 2.7,
+        "Reversal": 0.1,
+        "MeanPercent": -10,
+        "SDPercent": 1,
+    }
+    with caplog.at_level(logging.WARNING):
+        ringtest_stimulus_manager.interpret(target_onecell, stim_info)
+        assert "RelativeOrnsteinUhlenbeck signal is mostly zero" in caplog.text
```

Comment:
I know it is almost trivial but yous aid previously that the order of registering of the signals may be important. May you add a test where you check that the registering order is preserved against (with corner cases if they are there)

Reply:
Which order? The order of `SimConfig.stimuli` which determines the order of `self._stim_manager.interpret(target_spec, stim)` in node.py? There is one in `test_sonata_confg.py::test_parse_inputs`